pkg/certificate/providers/config.go

```diff
@@ -40,47 +42,65 @@ var getCA func(certificate.Issuer) (pem.RootCertificate, error) = func(i certifi
 }
 
 // NewCertificateManager returns a new certificate manager, with an MRC compat client.
-// TODO(4502): Use an informer behind a feature flag.
-func NewCertificateManager(kubeClient kubernetes.Interface, kubeConfig *rest.Config, cfg configurator.Configurator,
-	providerNamespace string, options Options, msgBroker *messaging.Broker) (*certificate.Manager, error) {
+// TODO(4713): Use an informer behind a feature flag.
+func NewCertificateManager(ctx context.Context, kubeClient kubernetes.Interface, kubeConfig *rest.Config, cfg configurator.Configurator,
+	providerNamespace string, options Options, msgBroker *messaging.Broker, ic *informers.InformerCollection, checkInterval time.Duration) (*certificate.Manager, error) {
 	if err := options.Validate(); err != nil {
 		return nil, err
 	}
 
-	// TODO(4502): Switch the compat client to an informer. Might need another struct to compose the informer and
-	// provider generator.
-	mrcClient := &MRCCompatClient{
-		MRCProviderGenerator: MRCProviderGenerator{
-			kubeClient:      kubeClient,
-			kubeConfig:      kubeConfig,
-			KeyBitSize:      cfg.GetCertKeyBitSize(),
-			caExtractorFunc: getCA,
-		},
-		mrc: &v1alpha2.MeshRootCertificate{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "legacy-compat",
-				Namespace: providerNamespace,
-				Annotations: map[string]string{
-					constants.MRCVersionAnnotation: "legacy-compat",
-				},
+	var mrcClient certificate.MRCClient
+	if ic == nil || len(ic.List(informers.InformerKeyMeshRootCertificate)) == 0 {
+		// no MRCs detected; use the compat client
```

Comment:
Right now a MRC is always created by osm-bootstrap.  If we want to support the option of not using the MRC created on startup to configure the Manager, maybe it would be best to make using the MRC an opt in option on install?

Reply:
I don't have strong opinions either way. If we ARE making a hard cut though, we're likely going to want some documentation on the change and instructions on how to switch